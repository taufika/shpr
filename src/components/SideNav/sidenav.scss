@import '../../variables/colors.scss';

.sidenav {
  background-color: #fff;
  padding: 12px 0;
  width: 240px;
  flex-shrink: 0;

  .nav-item {
    padding: 12px 40px 12px 20px;
    font-weight: bold;

    .fas, .far {
      margin-right: 12px;
    }

    &.active {
      color: $primary-color;
      border-left: 4px solid $primary-color;
      padding-left: 16px;
    }
  }
}