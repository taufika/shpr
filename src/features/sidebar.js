import { createSlice } from '@reduxjs/toolkit'

export const sidebarSlice = createSlice({
  name: 'users',
  initialState: {
    isSidebarActive: false,
  },
  reducers: {
    toggleSidebar: (state) => {
      state.isSidebarActive = !state.isSidebarActive;
    },
  }
});

export const { toggleSidebar } = sidebarSlice.actions;

export default sidebarSlice.reducer;