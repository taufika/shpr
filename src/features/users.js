import { createSlice } from '@reduxjs/toolkit'

export const usersSlice = createSlice({
  name: 'users',
  initialState: {
    users: [],
    status: 'ready',
  },
  reducers: {
    setLoading: (state) => {
      state.status = 'loading';
    },
    setReady: (state) => {
      state.status = 'ready';
    },
    setError: (state) => {
      state.status = 'error';
    },
    assignUsers: (state, action) => {
      state.users = action.payload;
    },
  }
});

export const { assignUsers, setError, setReady, setLoading } = usersSlice.actions;

export default usersSlice.reducer;