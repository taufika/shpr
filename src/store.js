import { configureStore } from '@reduxjs/toolkit'
import usersReducer from './features/users';
import sidebarReducer from './features/sidebar';

export default configureStore({
  reducer: {
    users: usersReducer,
    sidebar: sidebarReducer,
  },
})
