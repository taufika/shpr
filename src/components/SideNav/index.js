import React from 'react';

import './sidenav.scss';

export default class SideNav extends React.Component {
  render() {
    return <div className='sidenav'>
      <div className='nav-item'>Beranda</div>
      <div className='nav-item active'>Driver Management</div>
      <div className='nav-item'>Pickup</div>
    </div>;
  };
};
