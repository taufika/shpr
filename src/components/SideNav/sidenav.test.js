import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux'
import store from '../../store';
import SideNav from '.';
import { toggleSidebar } from '../../features/sidebar'

const wrapWithStore = () => {
  return <Provider store={store}><SideNav /></Provider>
}

describe('Component:SideNav', () => {
  it('Should match snapshot', () => {
    const { container } = render(wrapWithStore());
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('sidenav').classList).not.toContain('active');
  });

  it('should add active class on state change to sidebar active', () => {
    render(wrapWithStore());
    store.dispatch(toggleSidebar());
    expect(screen.getByTestId('sidenav').classList).toContain('active');
  });
});
