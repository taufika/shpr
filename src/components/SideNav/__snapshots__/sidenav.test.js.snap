// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:SideNav Should match snapshot 1`] = `
<div>
  <div
    class="sidenav "
    data-testid="sidenav"
  >
    <div
      class="nav-item"
    >
      <i
        class="fas fa-home"
      />
      <span>
        Beranda
      </span>
    </div>
    <div
      class="nav-item active"
    >
      <i
        class="fas fa-user"
      />
      <span>
        Driver Management
      </span>
    </div>
    <div
      class="nav-item"
    >
      <i
        class="far fa-calendar"
      />
      <span>
        Pickup
      </span>
    </div>
  </div>
</div>
`;
